### ISSUE TITLE
Problem with implementing hls4ml to a Tensorflow example 

### ISSUE THREAD

**Issue by ThGkasios** [fastmachinelearning/hls4ml#564]
Hello, I'm trying to test out hls4ml using one of TensorFlow's examples; picked up the [Simple audio recognition: Recognizing keywords](https://www.tensorflow.org/tutorials/audio/simple_audio) example, and appended code to the notebook for hls4ml.

To be able to run the featured model through `hls4ml`, I had to skip the `tf.keras.layers.Normalization` and `tf.keras.layers.Resizing` due to them not being supported by `hls4ml`.

After shaving quite a bit from the model layers so that Vivado can run the CSim without being stopped by a SIGSEGV error due to size, I have encountered the following problem: The results from the C Simulation come off as extremely off:

```
Processing input 0
Predictions
0.815873 0.00602767 0.0113112 0.0115806 0.00597961 0.13538 0.00994864 0.003899
Quantized predictions
-6.77641 -4.61245 9.64848 -12.4635 8.57524 -4.72453 -10.1261 9.31748
```
My current code for the model is 
```
base_model = models.Sequential([
    layers.Input(shape=input_shape),
    layers.Conv2D(4, 3, activation='relu'),
    layers.Conv2D(8, 3, activation='relu'),
    layers.MaxPooling2D(),
    layers.Dropout(0.25),
    layers.Flatten(),
    layers.Dense(16, activation='relu'),
    layers.Dropout(0.5),
    layers.Dense(num_labels), # this is always 8 for the output labels
])
```
and the rest of the code is unchanged from Tensorflow's example.

Is there something I'm missing?

**Comment by ThGkasios** [comment]
Hello, so, my problem with data being badly quantized is half-solved; raising the presicion to ap_fixed<32,20> seems to yield better results, as the hls_model's maximum output matches the maximum output from Tensorflow, which is enough for me to close this. When I manage to fix the rest of this I will append the solution.

**Comment by ThGkasios** [comment]
I have published my notebook with the code output [here](https://github.com/ThGkasios/hls4ml-tfsimpleaudio/blob/main/simple_audio.ipynb), for whomever is interested.
